provider/env.go

```diff
@@ -0,0 +1,42 @@
+package provider
+
+import (
+	"context"
+
+	"github.com/google/uuid"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+func envResource() *schema.Resource {
+	return &schema.Resource{
+		Description: "Use this resource to set an environment variable in a workspace.",
+		CreateContext: func(_ context.Context, rd *schema.ResourceData, _ interface{}) diag.Diagnostics {
+			rd.SetId(uuid.NewString())
```

Comment:
I considered disallowing envs here, like `PATH`, `CODER_AGENT_TOKEN`, etc. However, I opted for doing this agent side instead (we'll log an error).